src/window.ts

```diff
@@ -262,8 +262,25 @@ export class Window {
             thisCell.yoff = thatCell.yoff
             thisCell.fontSize = thatCell.fontSize
             if (thisCell.t) {
-                thisCell.t.options.fontSize = thatCell.fontSize * this.gate.fontScale
-                thisCell.fit()
+                const hasFraction = String(thatCell.fontSize * this.gate.fontScale).includes('.')
+                thisCell.t.options.fontSize = Math.floor(thatCell.fontSize * this.gate.fontScale) + (hasFraction ? .5 : 0)
+                if (thisCell.fitAddon.proposeDimensions) {
+                    const dims = thisCell.fitAddon.proposeDimensions()
+                    if (dims && (dims.cols !== thatCell.cols || dims.rows !== thatCell.rows)) {
+                        thisCell.t.options.fontSize -= .5
+                    }
+                }
+                thisCell.t._core._renderService.clear()
```

Comment:
Why clear? It's a bad idea to use `_core` as it's internal and we have no guarantee and it'll work. It also doesn't work on safari.